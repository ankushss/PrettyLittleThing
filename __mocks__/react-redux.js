module.exports = {
  useDispatch: jest.fn(() => jest.fn()),
  useSelector: jest.fn(),
};
