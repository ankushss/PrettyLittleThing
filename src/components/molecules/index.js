export * from './DressItem';
export * from './CartItem';
