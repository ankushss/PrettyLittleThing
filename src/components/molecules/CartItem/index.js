import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import {useDispatch} from 'react-redux';
import {addToCart, removeFromCart} from '../../../store/cart';
import {Counter} from '../../atoms';

export const CartItem = ({item}) => {
  const dispatch = useDispatch();

  const onPressAdd = () => {
    dispatch(addToCart(item));
  };

  const onPressRemove = () => {
    dispatch(removeFromCart(item));
  };

  return (
    <View style={styles.container}>
      <Image source={{uri: item.img}} style={styles.image} />
      <View style={styles.textContainer}>
        <Text numberOfLines={2}>{item.name}</Text>
        <Text style={styles.price}>${item.price}</Text>
        <Text>
          Color: <Text style={styles.colorText}>{item.colour}</Text>
        </Text>
        <View style={styles.counterContainer}>
          <Counter
            value={item.quantity}
            onPressAdd={onPressAdd}
            onPressRemove={onPressRemove}
          />
          <Text>
            Total:{' '}
            <Text style={styles.totalText}>${item.quantity * item.price}</Text>
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  image: {
    width: 80,
    height: (1180 / 740) * 80,
    resizeMode: 'contain',
    marginRight: 10,
  },
  textContainer: {
    flex: 1,
  },
  price: {
    fontWeight: 'bold',
    fontSize: 20,
    marginTop: 4,
  },
  colorText: {
    fontWeight: 'bold',
  },
  button: {
    marginTop: 10,
  },
  counterContainer: {
    marginTop: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  totalText: {
    fontWeight: 'bold',
  },
});
