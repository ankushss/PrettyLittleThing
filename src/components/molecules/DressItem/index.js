import React from 'react';
import {Image, StyleSheet, Text, useWindowDimensions, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {addToCart, getItemQuantityInCart} from '../../../store/cart';

import {Button} from '../../atoms';

export const DressItem = ({item, index}) => {
  const dimensions = useWindowDimensions();
  const dispatch = useDispatch();
  const itemQuantityInCart = useSelector(getItemQuantityInCart(item.id));
  const isItemAddedToCard = itemQuantityInCart > 0;

  const onPressAddToCard = () => {
    dispatch(addToCart(item));
  };

  return (
    <View style={styles.container(dimensions, index)}>
      <Image source={{uri: item.img}} style={styles.image(dimensions)} />
      <Text numberOfLines={2}>{item.name}</Text>
      <Text style={styles.price}>${item.price}</Text>
      <Text>
        Color: <Text style={styles.colorText}>{item.colour}</Text>
      </Text>
      <Button
        testID="addToCartButton"
        variant={isItemAddedToCard ? 'outlined' : 'contained'}
        style={styles.button}
        onPress={onPressAddToCard}
        disabled={isItemAddedToCard}>
        {isItemAddedToCard ? 'ADDED TO CART' : 'ADD TO CART'}
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: (dimensions, index) => ({
    width: (dimensions.width - 60) / 2,
    marginRight: index % 2 === 0 ? 20 : 0,
    marginBottom: 20,
  }),
  image: dimensions => ({
    width: (dimensions.width - 60) / 2,
    height: ((1180 / 740) * (dimensions.width - 60)) / 2,
    resizeMode: 'contain',
    marginBottom: 10,
  }),
  price: {
    fontWeight: 'bold',
    fontSize: 20,
    marginTop: 4,
  },
  colorText: {
    fontWeight: 'bold',
  },
  button: {
    marginTop: 10,
  },
});
