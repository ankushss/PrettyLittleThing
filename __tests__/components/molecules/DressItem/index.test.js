import React from 'react';
import * as redux from 'react-redux';
import {create} from 'react-test-renderer';
import {DressItem} from '../../../../src/components/molecules';

const mockUseSelector = jest.spyOn(redux, 'useSelector');

const mockItem = {
  id: 1,
  colour: 'mockColor',
  name: 'mockName',
  price: 1,
  img: 'mockImgUrl',
};

describe('DressItem', () => {
  it('matches snapshot', () => {
    const component = create(<DressItem item={mockItem} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('onPress AddToCart', () => {
    mockUseSelector.mockReturnValueOnce(2);
    const component = create(<DressItem item={mockItem} />);
    const button = component.root.findByProps({testID: 'addToCartButton'});
    button.props.onPress();
  });
});
