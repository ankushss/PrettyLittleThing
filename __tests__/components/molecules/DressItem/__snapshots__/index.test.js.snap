// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DressItem matches snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "marginRight": 0,
      "width": 345,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "mockImgUrl",
      }
    }
    style={
      Object {
        "height": 550.1351351351351,
        "marginBottom": 10,
        "resizeMode": "contain",
        "width": 345,
      }
    }
  />
  <Text
    numberOfLines={2}
  >
    mockName
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 4,
      }
    }
  >
    $
    1
  </Text>
  <Text>
    Color: 
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      mockColor
    </Text>
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginTop": 10,
        "opacity": 1,
      }
    }
    testID="addToCartButton"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#00aa5b",
            "borderColor": "#00aa5b",
            "borderWidth": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        ADD TO CART
      </Text>
    </View>
  </View>
</View>
`;
